### PR TITLE
[JB][SUPP-161] add confirm dd details page

### PR DESCRIPTION
**Pull request by barnesjake**

### CODE REVIEW

File: app/controllers/BankDetailsController.scala [hunk 1]
```diff
@@ -98,7 +98,19 @@ class BankDetailsController @Inject() (
   }
 
   val checkBankDetails: Action[AnyContent] = as.eligibleJourneyAction { implicit request =>
-    Ok("This is where the check bank details page will go, for now, here's the journey:\n\n\n" + Json.prettyPrint(Json.toJson(request.journey)))
+    request.journey match {
+      case j: Journey.BeforeEnteredDirectDebitDetails => JourneyIncorrectStateRouter.logErrorAndRouteToDefaultPage(j)
+      case j: Journey.AfterEnteredDirectDebitDetails  => displayCheckBankDetailsPage(j)
```
Comment (AndyHWChung) [inline app/controllers/BankDetailsController.scala:103]: we could put an extra guard here on the stage to see if the user said that they were the sole signatory - if they said no then we shouldn't show this page

File: test/controllers/BankDetailsControllerSpec.scala [hunk 2]
```diff
@@ -220,12 +237,64 @@ class BankDetailsControllerSpec extends ItSpec {
...
+      val summaries = doc.select(".govuk-summary-list").select(".govuk-summary-list__row").iterator().asScala.toList
+      summaries.size shouldBe 3
+
+      val expectedAccountNameRow = SummaryRow("Account name", "Bob Ross", routes.BankDetailsController.enterBankDetails().url)
+      val expectedSortCodeRow = SummaryRow("Sort code", "123456", routes.BankDetailsController.enterBankDetails().url)
+      val expectedAccountNumberRow = SummaryRow("Account number", "12345678", routes.BankDetailsController.enterBankDetails().url)
+      val expectedSummaryRows = List(expectedAccountNameRow, expectedSortCodeRow, expectedAccountNumberRow)
+      extractSummaryRows(summaries) shouldBe expectedSummaryRows
+
+      //assert direct debit content
```
Comment (AndyHWChung) [inline test/controllers/BankDetailsControllerSpec.scala]: is this something you forgot to do or intended to do in the future?

File: app/controllers/BankDetailsController.scala (hunk 1 above)
Comment (barnesjake) [reply app/controllers/BankDetailsController.scala:103]: Sure, will update

File: test/controllers/BankDetailsControllerSpec.scala (hunk 2 above)
Comment (barnesjake) [reply test/controllers/BankDetailsControllerSpec.scala]: Yeah, I forgot to do this but already updated in most recent commit

**Review by nubz — APPROVED** [review]

File: app/controllers/BankDetailsController.scala (hunk 1 above)
Comment (barnesjake) [reply app/controllers/BankDetailsController.scala]: Done, I added a test for this too

File: app/controllers/BankDetailsController.scala [hunk 3]
```diff
@@ -98,7 +98,20 @@ class BankDetailsController @Inject() (
...
+      case j: Journey.AfterEnteredDirectDebitDetails =>
+        if (j.directDebitDetails.isAccountHolder) {
+          Ok(views.bankDetailsSummary(j.directDebitDetails, BankDetailsController.enterBankDetailsUrl))
+        } else {
+          Redirect(routes.BankDetailsController.cannotSetupDirectDebitOnlinePage().url)
+        }
+    }
+  }
+
+  val checkBankDetailsSubmit: Action[AnyContent] = as.eligibleJourneyAction.async { implicit request =>
```
Comment (AndyHWChung) [inline app/controllers/BankDetailsController.scala:112]: we should probably put that check on the stage on the submit endpoint as well - it'd be bad if we allowed someone to use this POST endpoint to bypass that check and proceed anyway. Not sure if anyone actually would but better to be safe

Comment (barnesjake) [reply app/controllers/BankDetailsController.scala:112]: will raise a ticket, I assume you mean in the backend?

Comment (barnesjake) [reply app/controllers/BankDetailsController.scala:112]: we'll probably be able to bring it into this sprint tbh

Comment (AndyHWChung) [reply app/controllers/BankDetailsController.scala:112]: no I meant around the bit of code before we talk to the backend here

Comment (barnesjake) [reply app/controllers/BankDetailsController.scala:112]: Done

**Review by AndyHWChung — APPROVED** [review]

**Pull request merged**